app/Http/Controllers/StudentController.php

```diff
@@ -163,43 +163,39 @@ private function updateCreditsCompleted(student $student): void
         $creditsCompletedFirstYear = 0.0;
         $electivesCompletedSecondYear = 0.0;
 
-        // if any courses are completed
-        if ($student->completedCourses && $student->completedCourses->course) {
```

Comment:
Is it safe to remove this check completely?

Reply:
`$student->completedCourses` is replaced with `$student->completedCoursesV2` so that check is now unnecessary.

And because `$student->completedCoursesV2` is a `Illuminate\Database\Eloquent\Collection`, it is not nullable, which means that we don't need to check for it's presence. If the student does not have any completed courses, the collection will be empty, but not null.

Reply:
I distinctly remember writing the fillChildren() method because none of the attached models worked until they explicitly had something attached to them. Later, I realized that I also had to refresh the parent model before they would work.

Regardless, fillChildren() should be done during creation, if it is still necessary, so I'll accept this and treat that as a sperate issue.